packages/core/src/routes/verification/index.ts

```diff
@@ -0,0 +1,55 @@
+import { VerificationType } from '@logto/schemas';
+import { generateStandardId } from '@logto/shared';
+import { z } from 'zod';
+
+import koaGuard from '#src/middleware/koa-guard.js';
+
+import koaOidcAuth from '../../middleware/koa-auth/koa-oidc-auth.js';
+import { expirationTime } from '../../queries/verification-records.js';
+import type { ProfileRouter, RouterInitArgs } from '../types.js';
+
+import { verificationRecordResponseGuard } from './types.js';
+import { transformVerificationRecord } from './utils.js';
+
+export default function verificationRoutes<T extends ProfileRouter>(
+  ...[router, { provider, queries, libraries }]: RouterInitArgs<T>
+) {
+  const {
+    verificationRecords,
+    users: { findUserById },
+  } = queries;
+  const {
+    users: { verifyUserPassword },
+  } = libraries;
+
+  router.use(koaOidcAuth(provider));
+
+  router.post(
+    '/verifications/password',
+    koaGuard({
+      body: z.object({ password: z.string().min(1) }),
+      response: verificationRecordResponseGuard,
+      status: [201, 422],
+    }),
+    async (ctx, next) => {
+      const { id: userId } = ctx.auth;
+      const { password } = ctx.guard.body;
+
+      const user = await findUserById(userId);
+      await verifyUserPassword(user, password);
+      const record = await verificationRecords.insert({
+        id: generateStandardId(),
+        type: VerificationType.Password,
+        userId,
+        expiresAt: new Date(Date.now() + expirationTime).valueOf(),
+        // Password verification is verified immediately
+        verifiedAt: Date.now(),
+      });
```

Comment:
Can we reuse the verification classes defined in the experience routes? Extract those classes to a shared place.
`/me/verifications` endpoints should have very much the same logic as experience verification endpoints.

Reply:
updated